src/iuse.cpp

```diff
@@ -5964,20 +5964,19 @@ bool einkpc_download_memory_card(player *p, item *eink, item *mc)
 
         std::vector<const recipe *> candidates;
 
-        for( auto &elem : recipe_dict ) {
+        for( const auto &e : recipe_dict ) {
+            const auto r = e.second;
 
-            const int dif = ( elem )->difficulty;
-
-            if (science) {
-                if( elem->valid_learn() ) {
-                    if (dif >= 3 && one_in(dif + 1)) {
-                        candidates.push_back( elem );
+                if (science) {
+                if( r.valid_learn() ) {
+                    if (r.difficulty >= 3 && one_in(r.difficulty + 1)) {
```

Comment:
Why not? Two nested clauses:
```c++
if( a ) {
    if( b ) {
      if( c ) {
          ...
      }
    }
} else {
    ...
}
```

are equivalent to one nested clause:

```c++
if( a ) {
    if( b && c ) {
      ...    
    }
} else {
    ...
}
```

or no nested clauses (preferable):

```c++
if( !a ) {
    ...    
} else if( b && c ) {
    ...
}
```